/*
 * This file was automatically generated by EvoSuite
 * Fri Apr 07 10:56:18 KST 2017
 */

package org.prevayler.foundation.network;

import static org.evosuite.runtime.EvoAssertions.assertThrownBy;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.fail;

import java.io.IOException;

import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.TooManyResourcesException;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVNET = true) 
public class NetworkServerObjectReceiverImpl_ESTest4 extends NetworkServerObjectReceiverImpl_ESTest_scaffolding4 {

  @Test
  public void test0()  throws Throwable  {
      NetworkServerObjectReceiverImpl networkServerObjectReceiverImpl0 = new NetworkServerObjectReceiverImpl((NetworkReceiverFactory) null, (Service) null, (ObjectServerSocket) null);
      // Undeclared exception!
      try { 
        networkServerObjectReceiverImpl0.run();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test
  public void test1()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      NetworkServerObjectReceiverImpl networkServerObjectReceiverImpl0 = null;
      try {
        networkServerObjectReceiverImpl0 = new NetworkServerObjectReceiverImpl((NetworkReceiverFactory) networkImpl0, (Service) null, (-1));
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Port value out of range: -1
         //
         assertThrownBy("org.evosuite.runtime.mock.java.net.MockServerSocket", e);
      }
  }

  @Test
  public void test2()  throws Throwable  {
      StubbornNetworkImpl stubbornNetworkImpl0 = new StubbornNetworkImpl();
      stubbornNetworkImpl0.createService((Service) null, 1557);
      NetworkServerObjectReceiverImpl networkServerObjectReceiverImpl0 = null;
      try {
        networkServerObjectReceiverImpl0 = new NetworkServerObjectReceiverImpl((NetworkReceiverFactory) stubbornNetworkImpl0, (Service) null, 1557);
        fail("Expecting exception: IOException");
      
      } catch(IOException e) {
         //
         // Failed to open TCP port
         //
         assertThrownBy("org.evosuite.runtime.mock.java.net.EvoSuiteSocket", e);
      }
  }

  @Test
  public void test3()  throws Throwable  {
      StubbornNetworkImpl stubbornNetworkImpl0 = new StubbornNetworkImpl();
      NetworkServerObjectReceiverImpl networkServerObjectReceiverImpl0 = new NetworkServerObjectReceiverImpl((NetworkReceiverFactory) stubbornNetworkImpl0, (Service) null, 0);
      networkServerObjectReceiverImpl0.shutdown();
      assertTrue(networkServerObjectReceiverImpl0.isDaemon());
  }

  @Test
  public void test6()  throws Throwable  {
      NetworkImpl networkImpl0 = new NetworkImpl();
      NetworkServerObjectReceiverImpl networkServerObjectReceiverImpl0 = new NetworkServerObjectReceiverImpl(networkImpl0, (Service) null, (ObjectServerSocket) null);
      // Undeclared exception!
      try { 
        networkServerObjectReceiverImpl0.shutdown();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("org.prevayler.foundation.network.NetworkServerObjectReceiverImpl", e);
      }
  }
}
